src/ControllerMethod.php

```diff
@@ -95,25 +100,77 @@ static function parse(string $context, array $definitions, ClassMethod $method,
 		}
 
 		foreach ($methodParameters as $methodParameter) {
-			$param = null;
 			$methodParameterName = $methodParameter->var->name;
 
-			foreach ($docParameters as $docParameter) {
-				$docParameterName = substr($docParameter->parameterName, 1);
-
-				if ($docParameterName == $methodParameterName) {
-					$param = new ControllerMethodParameter($context, $definitions, $methodParameterName, $methodParameter, $docParameter);
-					break;
+			$paramTag = null;
+			$psalmParamTag = null;
+			foreach ($docParameters as $docParameterType => $typeDocParameters) {
+				foreach ($typeDocParameters as $docParameter) {
+					$docParameterName = substr($docParameter->parameterName, 1);
+
+					if ($docParameterName == $methodParameterName) {
+						if ($docParameterType == "@param") {
+							$paramTag = $docParameter;
+						} else if ($docParameterType == "@psalm-param") {
+							$psalmParamTag = $docParameter;
+						} else {
+							Logger::panic($context, "Unknown param type " . $docParameterType);
+						}
+					}
 				}
 			}
 
-			if ($param == null) {
-				if ($allowMissingDocs) {
-					$param = new ControllerMethodParameter($context, $definitions, $methodParameterName, $methodParameter, null);
+			if ($paramTag !== null && $psalmParamTag !== null) {
+				// Use all the type information from @psalm-param because it is more specific,
+				// but pull the description from @param and @psalm-param because usually only one of them has it.
+				if ($psalmParamTag->description !== "") {
+					$description = $psalmParamTag->description;
+				} else if ($paramTag->description !== "") {
+					$description = $paramTag->description;
 				} else {
-					Logger::error($context, "Missing doc parameter for '" . $methodParameterName . "'");
-					continue;
+					$description = "";
 				}
+
+				try {
+					$type = OpenApiType::resolve(
+						$context,
+						$definitions,
+						new ParamTagValueNode(
+							$psalmParamTag->type,
+							$psalmParamTag->isVariadic,
+							$psalmParamTag->parameterName,
+							$description,
+							$psalmParamTag->isReference,
+						),
+					);
+				} catch (LoggerException $e) {
+					Logger::warning($context, "Unable to parse parameter " . $methodParameterName . ": " . $e->message . "\n" . $e->getTraceAsString());
```

Comment:
```suggestion
					Logger::warning($context, "Unable to parse parameter " . $methodParameterName . ": " . $e->message);
```

The stack is fully inside the extractor code and therefor not helpful for the user/developer trying to generate api. So I would remove it.

Reply:
`generate-spec --debug` or `generate-spec --verbose` incoming :)

Reply:
Ok ;)